### PR TITLE
Create a rector set for Doctrine DBAL 4.0

### PR DESCRIPTION
**Pull request by keulinho**
First set of rector rules for doctrine dbal 4.0 breaks, refer to https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md

### CODE REVIEW

**Comment by samsonasik** [comment]
@keulinho please run the following command:

```
vendor/bin/rector && composer fix-cs
```

To fix code anlayze notice.

@TomasVotruba the ECS pending seems can be removed from required workflow on CI:

```bash
 code_analysis / ECS Expected — Waiting for status to be reported 
```

**Comment by TomasVotruba** [comment]
@samsonasik Done :+1:

File: config/sets/doctrine-dbal-40.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Rector\Renaming\Rector\ClassConstFetch\RenameClassConstFetchRector;
+use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
+use Rector\Renaming\ValueObject\MethodCallRename;
+use Rector\Renaming\ValueObject\RenameClassAndConstFetch;
+
+return static function (RectorConfig $rectorConfig): void {
+    // https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-connectionparam__array-constants
```
Comment (johanadivare) [inline config/sets/doctrine-dbal-40.php:12]:
```suggestion
    // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-connectionparam__array-constants    
```

File: config/sets/doctrine-dbal-40.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            new RenameClassAndConstFetch(
+                'Doctrine\DBAL\Connection',
+                'PARAM_STR_ARRAY',
+                'Doctrine\DBAL\ArrayParameterType',
+                'STRING'
+            )
+        ]
+    );
+  
+    // https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-connection_schemamanager-and-connectiongetschemamanager
```
Comment (johanadivare) [inline config/sets/doctrine-dbal-40.php:25]:
```suggestion
    // @see https://github.com/doctrine/dbal/blob/4.0.x/UPGRADE.md#bc-break-removed-connection_schemamanager-and-connectiongetschemamanager
```

File: config/sets/doctrine-dbal-40.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+declare(strict_types=1);
+
+use Rector\Config\RectorConfig;
+use Rector\Renaming\Rector\ClassConstFetch\RenameClassConstFetchRector;
+use Rector\Renaming\Rector\MethodCall\RenameMethodRector;
+use Rector\Renaming\ValueObject\MethodCallRename;
+use Rector\Renaming\ValueObject\RenameClassAndConstFetch;
+
+return static function (RectorConfig $rectorConfig): void {
```
Comment (johanadivare) [inline config/sets/doctrine-dbal-40.php:11]: Please define a constant in `DoctrineSetList.php`

**Comment by keulinho** [comment]
@samsonasik i fixed the ECS issues, but not sure if I also should commit the rector changes as those are completely unrelated to my PR :thinking:

**Comment by samsonasik** [comment]
CI needs  green to make it merge-able, I can take care of that :)

**Comment by samsonasik** [comment]
Thank you @keulinho

**Pull request merged**